File: problem55.py
def plus_reverse(n):
    s=str(n)
    s_rev=str()
    for i in range(0,len(s)):
        s_rev+=s[len(s)-1-i]
    n_rev=int(s_rev)
    return n+n_rev
def palindromic(n):
    s=str(n)
    for i in range(0,int(len(s)/2)):
        if(s[i]!=s[len(s)-1-i]):
            return False
    return True
result=0
for i in range(1,10000):
    number=i
    x=0
    while(x<50):
        number=plus_reverse(number)
        x+=1
        if(palindromic(number)):
            break
    if(x==50 and palindromic(number)==False):
        result+=1
print(result)
x=input()

File: problem55.cpp
#include <bits/stdc++.h>
using namespace std;
unsigned long long int plus_reverse(unsigned long long int i)
{
	unsigned long long int n=i,m=0;
	while(n>0)
	{
		m=10*m+n%10;
		n/=10;
	}
	return m+i;
}
bool palindromic(unsigned long long int x)
{
	unsigned long long int y=1,n=x;
	while(n!=0)
	{
		y*=10;
		n/=10;
	}
	y/=10;
	n=10;
	while(true)
	{
		if((x/y)%10!=(x%n)/(n/10)){return false;}
		if(y<=n){return true;}
		y/=10;
		n*=10;
	}
}
int main()
{
	int result=0;
	for(int i=1;i<10000;i++)
	{
		unsigned long long int number=i;
		int x=0;
		while(x<50)
		{
			number=plus_reverse(number);
			x++;
			if(palindromic(number)){break;}
			if(x==50 && !palindromic(number)){result++;}
		}
	}
	cout<<result;
	getchar();
	return 0;
}
